### DISCUSSION TITLE
How can I exclude relative domain URLs from using the `rehype-external-links' plugin?

### DISCUSSION THREAD

**Discussion by HenestrosaDev** [rehypejs/rehype#153] [Q&A] [answered]
Hello.

I'm using Astro and I want to exclude the use of the `rehype-external-links' plugin for URLs relative to the site's domain. My current implementation of the plugin in the Astro configuration is as follows:

```js
rehypePlugins: [
  [
    rehypeExternalLinks,
    {
      target: "_blank",
      rel: "noreferrer",
    },
  ],
],
```

**Comment by wooorm** [comment]
The plugin does what you ask. Perhaps you can use more words to describe your question.

**Reply by HenestrosaDev** [reply]
For example, I have my site henestrosaconh.com. How can I prevent the plugin from applying `target: "_blank"` and `rel: "noreferrer"` to the URLs of said domain?

**Reply by wooorm** [reply, marked as answer]
The plugin applies to absolute URLs, so those with `http:...`. You can write relative URLs, so like say `[text](/some/url/whatever)`? Why are you writing absolute URLs?

**Reply by wooorm** [reply]
If you‘re looking for more complex tests, perhaps use the `test` option, see the readme: https://github.com/rehypejs/rehype-external-links

**Reply by HenestrosaDev** [reply]
You're right. Sorry for the confusion, I thought I was using relative URLs instead of absolute URLs. I've changed them to relative and now they work as expected. Thanks so much! :)

**Reply by wooorm** [reply]
P.S. there’s a plugin that does that can do that automatically: https://github.com/rehypejs/rehype-minify/tree/main/packages/rehype-minify-url#unifieduserehypeminifyurl-options
